### PR TITLE
Viral Wild adaption symptoms for rat disease and level 7 virus outbreak

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Rat virus / level 7 viral outbreak now is stronger.
These get a random bonus virus symptom that even can go 7th symptom.

## Why It's Good For The Game
natural viruses are mostly weak and goes nothing in most time.
even if level 7 virus outbreak happens, nobody cares, as everyone knows it's nothing.
now I added new symptoms that can make natural viruses strong.
```
// high resistance
/datum/symptom/wild/wildresistance
	name = "Viral Wild-Resistance Adaption"
	stealth = -2
	resistance = 7
	stage_speed = 2
	transmission = -1

// high stage speed
/datum/symptom/wild/wildspeed
	name = "Viral Wild-Speed Adaption"
	stealth = 0
	resistance = 2
	stage_speed = 10
	transmission = -1

// High stealth
/datum/symptom/wild/wildstealth
	name = "Viral Wild-Stealth Adaption"
	stealth = 7
	resistance = 2
	stage_speed = 2
	transmission = -7

// High tranmission, minus stealth
/datum/symptom/wild/wildtransmission
	name = "Viral Wild-Transmission Adaption"
	stealth = -20
	resistance = 3
	stage_speed = 3
	transmission = 10

// when it failed to get any good stat from random code
/datum/symptom/wild/wildbrutality
	name = "Viral Wild-Brutality Adaption"
	stealth = -20
	resistance = 15
	stage_speed = 15
	transmission = 15 // this is not transmitible. no worries.
```
(Note: brutality has high transmission, but it is not contagious. It's only for enabling transmission threshold.)

These only appear from rats or level 7 virus outbreak, boosting stats hightly, and takes 7rd symptom slots

You might think stats are a bit high, but it's intentional, and it was actually pretty fine to virus mostly.
I uploaded screenshots below, so check it please.

Accepting a wild symptom has a condition, like being `Resistance >=`
below is how the portion of each wild symptom would spawn

Calculation-
Samples: Mouse spawn 6,000 mobs
* Resistance >=5, `26%` (1563 samples)
* Stage Speed >= 2, `26%` (1543 samples)
* Stealth >=0, `25%` (1506 samples)
* Transmission >=0, `11%` (638 samples) - average 2 transmission
* Fail to pass any above, `12%` (750 samples) - total stat average -9.94

Samples: random culture 4,000 spawns
* Resistance >=5, `35%` (1409 samples)
* Stage Speed >=2, `25%` (986 samples)
* Stealth >=0, `19%` (779 samples)
* Tranmission >=0, `8%` (316 samples)
* Fail to pass any above, `13%` (510 samples) - total stat average -11.98

Mostly, it would only fuck one, otherwise, at 11%~8% chance, the virus becomes airborne. so It's still rare.

---------------------

only 15% rats get wild symptom.
if you have 100 rats in the station, 75% rats get diseases, and 20% of the rats get a wild symptom.
while level 7 virus outbreak has 100% chance to have a wild symptom.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

First, virology rat is a hamster that doesn't have a wild virus.

--------------

image

This is from a rat. it has a wild virus, but not detectable.

--------------

image

Let's say it's detectable anyway

--------------
image
image
image
image

examples. actually pretty fine.

--------------

image

if you try to edit a wild adaptation virus, it disappears.

--------------

image

Can you even guess it was actually a good virus?

</details>

## Changelog
:cl:
refactor: proof coding that a rat won't have a null disease
add: Wild-adaption virus symptoms. These only exist rarely in mice, or 100% appears from level 7 virus outbreak. these virus symptoms do nothing, but have extremely high stats (i.e. Resistance 15). 
/:cl:

### CODE REVIEW

File: code/datums/diseases/advance/advance.dm [hunk 1]
```diff
@@ -426,33 +425,76 @@
...
-	if(symptoms.len >= VIRUS_SYMPTOM_LIMIT)
-		RemoveSymptom(pick(symptoms))
+	if(symptoms.len >= VIRUS_SYMPTOM_LIMIT+VIRUS_WILD_COUNT)
+		var/datum/symptom/target
+		for(var/count in 1 to 150)
+			target = pick(symptoms)
+			if(!istype(target, /datum/symptom/wild))
+				break
+			if(count == 150)
+				return
```
Comment (EvilDragonfiend) [inline code/datums/diseases/advance/advance.dm:452]: I know this code is not beautiful, but trying a pick from a list without a specific object was not possible to me.

**Comment by zeskorion** [comment]
random virus outbreaks shouldnt be more powerful. but if you want them to be, don't execute it like this. this is super jank and is BEGGING for virologists to discover a glitch to use these symptoms for themselves

instead, set a new var for advance diseases, that, if set, sets a minimum spread transmission value, that overrides the calcs in refresh()

**Comment by zeskorion** [comment]
> random virus outbreaks shouldnt be more powerful. but if you want them to be, don't execute it like this. this is super jank and is BEGGING for virologists to discover a glitch to use these symptoms for themselves
> 
> instead, set a new var for advance diseases, that, if set, sets a minimum spread transmission value, that overrides the calcs in refresh()

**Comment by FortSumter** [comment]
I once ate a rat that gave me a disease for fire sweat and lube sweat that was so stealthy that only a blood test could reveal it. I'd say that the rat diseases are deadly enough.

**Comment by EvilDragonfiend** [comment]
> I once ate a rat that gave me a disease for fire sweat and lube sweat that was so stealthy that only a blood test could reveal it. I'd say that the rat diseases are deadly enough.

Not all of rat disease get this. Rat disease appear 75% and these get lethalier when pass 20% again. So, calculatively, ot is 15% chance to be critical rat disease.

**Comment by EvilDragonfiend** [comment]
I feel touching `refresh()` makes a parameter unnecessarily bloated.

**Comment by EvilDragonfiend** [comment]
Okay, I will think more, and write here again if it doesn't go well.

**Comment by EvilDragonfiend** [comment]
Okay, as far as I took a look at it, the jank that zams said is necessary. It should be what it is. instead they suggested, I wrote more cheat proof ones. what zams suggest isn't capable of removing the jank they said.

**Comment by zeskorion** [comment]
> > random virus outbreaks shouldnt be more powerful. but if you want them to be, don't execute it like this. this is super jank and is BEGGING for virologists to discover a glitch to use these symptoms for themselves
> > instead, set a new var for advance diseases, that, if set, sets a minimum spread transmission value, that overrides the calcs in refresh(), and make modification of the disease turn that off. Make random spawn diseases spawn with that var at a random amount. Shouldnt be turned off by disease mutation
> 
> I am trying to figure how it would go, bit using refresh() is not going well to think better. Hmmm.

are you sure it won't?
image
image
image
image
image

there's the basic code to make it work in advance.dm, should be able to just make wild diseases edit that specific var and it'll automatically apply. a lot less jank than this approach

**Comment by zeskorion** [comment]
note i dont condone this idea

**Comment by EvilDragonfiend** [comment]
hmm if you say so

**Comment by zeskorion** [comment]
i mean, im no maintainer

**Comment by EvilDragonfiend** [comment]
I know, but I believe you'd know how virus system works well than me and I am just respecting your opinion

**Pull request closed without merging**